spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/CategoricalColHashBucket.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+import com.intel.analytics.bigdl.tensor.{DoubleType, FloatType, Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+import scala.util.hashing.MurmurHash3
+
+/**
+ * CategoricalColHashBucket operation can convert feature string to a Sparse/Dense Tensor
+ *
+ * DenseTensor if transType = 0
+ * SparseTensor if  transType = 1
+ *
+ * Use this when your sparse/categorical features are in string or integer format
+ *
+ * This operation distributes your inputs into a finite number of buckets by hashing
+ *
+ * The Operation support the feature column with single-value or multi-value
+ *
+ * The output_id = Hash(input_feature_string) % hashBucketSize, ranging 0 to hashBucketSize-1
+ *
+ * The missing values in input Tensor can be represented by -1 for int and '''' for string
+ *
+ * @param hashBucketSize An Integer > 1. The number of buckets.
+ * @param strDelimiter The delimiter of feature string, default: ",".
+ * @param transType The type of output tensor, default: 1.
+ * @tparam T Numeric type. Parameter tensor numeric type. Only support float/double now
+ */
+
+class CategoricalColHashBucket[T: ClassTag](
+  val hashBucketSize: Int,
+  val strDelimiter: String = ",",
+  val transType: Int = 1
+  )(implicit ev: TensorNumeric[T])
+  extends Operation[Table, Tensor[T], T] {
+
+  output = Activity.allocate[Tensor[T], T]()
+
+  override def updateOutput(input: Table): Tensor[T] = {
```

Comment:
what about set the input type to Tensor[String]? For Integer, may be we can add a cast operation before the operation